test/e2e/Services/Docker/DockerContainer.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { DockerContainer } from '../../../../src/renderer/services/docker/DockerContainer';
+import { runCmdAsync } from '../../../../src/renderer/services/utils/cmd-utils';
+import { DockerCommand } from '../../../../src/renderer/services/docker/DockerCommand';
+
+class SimpleDockerContainer extends DockerContainer {
+    constructor() {
+        super({
+            image: 'alpine',
+            name: 'test-image',
+            detached: true,
+            cmd: 'tail -f /dev/null'
+        });
+    }
+}
+
+describe('docker container e2e tests', () => {
+    let dockerContainer: DockerContainer;
+
+    beforeEach(() => {
+        dockerContainer = new SimpleDockerContainer();
+    });
+
+    // clean up created container
+    afterEach(async () => {
+        const isRunning = await dockerContainer.isRunning();
+        if (isRunning) {
+            await dockerContainer.stop();
+        }
+        await runCmdAsync(`docker rm test-image`).catch();
+    }, 20000);
+
+    /**
+     * TEST CASE: run-check
+     * 1) run docker instance
+     * 2) check if instance is running using cmdRun
+     */
+    it('should execute test case: run-check', async () => {
+        if (await DockerContainer.isDockerInstalled()) {
+            await dockerContainer.run();
+            // wait for docker instance to start
+            while (!(await dockerContainer.isRunning())) {}
+
+            const res = await runCmdAsync(DockerCommand.ps('test-image', 'running'));
+            expect(res.stdout.split('\n')[1] !== '').toBeTruthy();
+        }
+    }, 10000);
+
+    /**
+     * TEST CASE run-stop-restart
+     * 1) run docker instance
+     * 2) check if instance is running
+     * 3) stop docker instance
+     * 4) check if instance is stopped
+     * 5) restart docker instance
+     * 6) check if instance is running
+     */
+    it('should execute test case: run-stop-restart', async () => {
+        if (await DockerContainer.isDockerInstalled()) {
+            // run docker instance
+            const docker = await dockerContainer.run();
+            while (!(await dockerContainer.isRunning())) {}
+            // check if instance is running
+            expect(await dockerContainer.isRunning()).toBeTruthy();
+            // stop docker instance
+            const stopInstance = await dockerContainer.stop();
+            // check if instance is stopped
+            expect(stopInstance).toBeTruthy();
+            expect(await dockerContainer.isRunning()).toBeFalsy();
+            // restart docker instance
+            const started = await dockerContainer.restart();
+            // check if instance is running
+            while (!(await dockerContainer.isRunning())) {}
+            expect(started).toBeTruthy();
+            expect(await dockerContainer.isRunning()).toBeTruthy();
+        }
+    }, 30000);
```

Comment:
if this test is longer than 10-15s just make it skip. So our CI doesnt take forever

Reply:
@mpetrunic I can't find a way to do this, maybe we just keep this test code but skip it?